File: spring-boot-server/src/main/kotlin/com/witkups/carsharing/users/user/AppUserRepository.java
package com.witkups.carsharing.users.user;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

public interface AppUserRepository extends JpaRepository<ApplicationUser, Long> {

  @Query("select u from ApplicationUser u where u.user.login = ?1")
  ApplicationUser getByUserLogin(String login);

  ApplicationUser findApplicationUserById(Long id);
}

File: spring-boot-server/src/main/kotlin/com/witkups/carsharing/users/user/AppUserRepository.kt
package com.witkups.carsharing.users.user

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query


interface AppUserRepository : JpaRepository<ApplicationUser, Long> {

  @Query("select u from ApplicationUser u where u.user.login = ?1")
  fun getByUserLogin(login: String): ApplicationUser

  fun findApplicationUserById(id: Long?): ApplicationUser
}
